nbsite/nbbuild.py

```diff
@@ -295,37 +298,98 @@ def render_notebook(nb_path, document, preprocessors=[]):
     doc = new_document(nb_path, document.settings)
 
     with open(nb_path, encoding='utf-8') as f:
-        text = f.read()
-
-    ntbk = nbformat.reads(text, as_version=NOTEBOOK_VERSION)
+        notebook_sourcestring = f.read()
 
-    for preprocessor in preprocessors:
-        ntbk, _ = preprocessor(ntbk, {})
+    # myst-nb has been entirely rewritten in 0.14 so we need two
+    # different code paths so handle this transition.
+    if Version(myst_nb.__version__) < Version('0.14'):
+        from myst_nb.parser import nb_to_tokens, nb_output_to_disc, tokens_to_docutils
 
-    md_parser, env, tokens = nb_to_tokens(
-        ntbk,
-        env.myst_config,
-        env.config["nb_render_plugin"],
-    )
+        ntbk = nbformat.reads(notebook_sourcestring, as_version=NOTEBOOK_VERSION)
 
-    # Delete rst temporarily to ensure 
-    rst_path = nb_path[:-5]+'rst'
-    if os.path.isfile(rst_path):
-        with open(rst_path) as f:
-            rst_text = f.read()
-        os.remove(rst_path)
+        for preprocessor in preprocessors:
+            ntbk, _ = preprocessor(ntbk, {})
+        md_parser, env, tokens = nb_to_tokens(
+            ntbk,
+            env.myst_config,
+            env.config["nb_render_plugin"],
+        )
+        # Delete rst temporarily to ensure 
+        rst_path = nb_path[:-5]+'rst'
+        if os.path.isfile(rst_path):
+            with open(rst_path) as f:
+                rst_text = f.read()
+            os.remove(rst_path)
+        else:
+            rst_text = None
+        nb_output_to_disc(ntbk, doc)
+        if rst_text is not None:
+            with open(rst_path, 'w') as f:
+                f.write(rst_text)
```

Comment:
@philippjfr I don't get what this was doing and why the notebook had to be saved to disc (to `/jupyter_execute` in practice). Do you remember why?